ding/policy/mbpolicy/mbsac.py

```diff
@@ -219,3 +221,197 @@ def _monitor_vars_learn(self) -> List[str]:
             'alpha',
             'target_q_value',
         ] + alpha_loss
+
+
+@POLICY_REGISTRY.register('stevesac')
+class STEVESACPolicy(SACPolicy):
+    """
+       Overview:
+           Model based SAC with stochastic value expansion (arXiv 1807.01675).\
+               This implementation also uses value gradient w.r.t the same STEVE target.
+
+       Config:
+           == ====================    ========    =============  =====================================
+           ID Symbol                  Type        Default Value  Description
+           == ====================    ========    =============  =====================================
+           1  ``learn.grad_clip`      float       100.0          | Max norm of gradients.
+           2  ``learn.ensemble_size`` int         1              | The number of ensemble world models.
+           == ====================    ========    =============  =====================================
+
+        .. note::
+            For other configs, please refer to ding.policy.sac.SACPolicy.
+       """
+
+    config = dict(
+        learn=dict(
+            # (float) Max norm of gradients.
+            grad_clip=100,
+            # (int) The number of ensemble world models.
+            ensemble_size=1,
+        )
+    )
+
+    @classmethod
+    def default_config(cls: type) -> EasyDict:
+        cfg = copy.deepcopy(cls.config)
+        cfg = EasyDict(deep_merge_dicts(super().config, cfg))
+        cfg.cfg_type = cls.__name__ + 'Dict'
+        return cfg
+
+    def _init_learn(self) -> None:
+        super()._init_learn()
+
+        self._grad_clip = self._cfg.learn.grad_clip
+        self._ensemble_size = self._cfg.learn.ensemble_size
+
+        self._target_model.requires_grad_(False)
+
+        # TODO: auto alpha
+        self._auto_alpha = False
+
+        def actor_fn(obs: Tensor):
+            obs, dim = fold_batch(obs, 1)
+            (mu, sigma) = self._learn_model.forward(obs, mode='compute_actor')['logit']
+            dist = Independent(Normal(mu, sigma), 1)
+            pred = dist.rsample()
+            action = torch.tanh(pred)
+
+            log_prob = dist.log_prob(
```

Comment:
why do this operation here, if it aims to correct tanh, why not use the implementation [here](https://github.com/opendilab/DI-engine/blob/main/ding/policy/sac.py#L819)

Reply:
ok, we need to unity this implementation in all the codes related to SAC in the future